MapboxCoreNavigation/RouteProgress.swift

```diff
@@ -296,401 +296,68 @@ open class RouteProgress {
-     */
-    public var intersectionDistances: Array<CLLocationDistance>?
-
-    /**
-     The distance in meters the user is to the next intersection they will pass through.
-     */
-    public var userDistanceToUpcomingIntersection: CLLocationDistance?
-
-    /**
-     Index into `step.instructionsDisplayedAlongStep` representing the current visual instruction for the step.
-     */
-    public var visualInstructionIndex: Int = 0
-
-    /**
-     An `Array` of remaining `VisualInstruction` for a step.
-     */
-    public var remainingVisualInstructions: [VisualInstructionBanner]? {
-        guard let visualInstructions = step.instructionsDisplayedAlongStep else { return nil }
-        return Array(visualInstructions.suffix(from: visualInstructionIndex))
-    }
-
-    /**
-     Index into `step.instructionsSpokenAlongStep` representing the current spoken instruction.
-     */
-    public var spokenInstructionIndex: Int = 0
-
-    /**
-     An `Array` of remaining `SpokenInstruction` for a step.
-     */
-    public var remainingSpokenInstructions: [SpokenInstruction]? {
-        guard let instructions = step.instructionsSpokenAlongStep else { return nil }
-        return Array(instructions.suffix(from: spokenInstructionIndex))
-    }
-
-    /**
-     Current spoken instruction for the user's progress along a step.
-     */
-    public var currentSpokenInstruction: SpokenInstruction? {
-        guard let instructionsSpokenAlongStep = step.instructionsSpokenAlongStep else { return nil }
-        guard spokenInstructionIndex < instructionsSpokenAlongStep.count else { return nil }
-        return instructionsSpokenAlongStep[spokenInstructionIndex]
-    }
-
-    /**
-     Current visual instruction for the user's progress along a step.
-     */
-    public var currentVisualInstruction: VisualInstructionBanner? {
-        guard let instructionsDisplayedAlongStep = step.instructionsDisplayedAlongStep else { return nil }
-        guard visualInstructionIndex < instructionsDisplayedAlongStep.count else { return nil }
-        return instructionsDisplayedAlongStep[visualInstructionIndex]
-    }
-    
-    public var keyPathsAffectingValueForRemainingVisualInstructions: Set<String> {
-        return ["step.instructionsDisplayedAlongStep", "visualInstructionIndex"]
+        let route = try container.decode(Route.self, forKey: .indexedRoute)
+        let routeIndex = try container.decode(Int.self, forKey: .indexedRouteIndex)
+        self.indexedRoute = (route, routeIndex)
+        self.routeOptions = try container.decode(RouteOptions.self, forKey: .routeOptions)
+        self.legIndex = try container.decode(Int.self, forKey: .legIndex)
+        self.currentLegProgress = try container.decode(RouteLegProgress.self, forKey: .currentLegProgress)
+        self.congestionTravelTimesSegmentsByStep = decodeCongestion(try container.decode([[[TimedCongestionLevelCodable]]].self,
```

Comment:
Even though `congestionTravelTimesSegmentsByStep` and `congestionTimesPerStep` can be recalculated using `calculateLegsCongestion()`, I still decided to explicitly encode/decode them because these are `public var` properties and can be potentially modified by User.

Reply:
I think it was an oversight that the setters are public. Nothing in MapboxNavigation appears to read or write to these properties. Should we privatize these properties or at least their setters in v1.0.0 to avoid the headache and size impact of encoding these ordinarily computed properties?

Reply:
I can barely imagine a case when user needs to manually adjust those properties, so IMHO we could at least make it `readonly`. However, such change better be deployed in `1.0` but this current PR is targeted for `1.1`...

Reply:
As long as the read-only change is in a separate commit or separate PR, we can cherry-pick it separately to the release-v1.0 branch.